plugin/serve.go

```diff
@@ -7,11 +7,15 @@ import (
 	"github.com/hashicorp/terraform/terraform"
 )
 
-// The constants below are the names of the plugins that can be dispensed
-// from the plugin server.
 const (
+	// The constants below are the names of the plugins that can be dispensed
+	// from the plugin server.
 	ProviderPluginName    = "provider"
 	ProvisionerPluginName = "provisioner"
+
+	// DefaultProtocolVersion is the default protocol version to configure
+	// clients and server
```

Comment:
Maybe we should be explicit here about the overriding behavior, since we've confused ourselves about it a bunch of times already!

```go
// DefaultProtocolVersion is the protocol version assumed for legacy clients that don't specify
// a particular version during their handshake. This is the version used when Terraform 0.10
// and 0.11 launch plugins that were built with support for both versions 4 and 5, and must
// stay unchanged at 4 until we intentionally build plugins that are not compatible with 0.10 and
// 0.11.
```

Reply:
🙏 this is much more clear, thank you for strengthening this context (and my understanding of where we stand!)